src/dataset-builder/DemographicsChart.ts

```diff
@@ -0,0 +1,160 @@
+import _ from 'lodash/fp';
+import { generateRandomNumbers, generateRandomNumbersThatAddUpTo } from 'src/dataset-builder/TestConstants';
+
+interface ChartLabel {
+  short: string;
+  long?: string;
+}
+export interface ChartSeries {
+  name?: string;
+  data: number[];
+}
+
+export interface CohortDemographics {
+  categories: ChartLabel[];
+  series: ChartSeries[];
+  title: string;
+  yTitle: string;
+  height: string;
+  legendEnabled: boolean;
+  showSeriesName: boolean;
+}
+export function chartOptions(cohortDemographics: CohortDemographics) {
+  return {
+    chart: {
+      spacingLeft: 20,
+      spacingRight: 30,
+      height: cohortDemographics.height,
+      style: { fontFamily: 'inherit' },
+      type: 'bar',
+    },
+    legend: { enabled: cohortDemographics.legendEnabled },
+    plotOptions: { series: { stacking: 'normal' } },
+    series: cohortDemographics.series,
+    title: {
+      align: 'left',
+      style: { fontSize: '16px', fontWeight: 'bold', color: '#333f52' },
+      text: cohortDemographics.title,
+    },
+    tooltip: {
+      followPointer: true,
+      formatter() {
+        // @ts-ignore
+        // eslint-disable-next-line react/no-this-in-sfc
+        const currCategory: ChartLabel = _.find((category) => category.short === this.x, cohortDemographics.categories);
+        const categoryDescription = currCategory.long || currCategory.short;
+        if (cohortDemographics.showSeriesName) {
+          // @ts-ignore
+          // eslint-disable-next-line react/no-this-in-sfc
+          return `${categoryDescription} <br/><span style="color:${this.color}">\u25CF</span> ${this.series.name}<br/> ${this.y}`;
+        }
+        // @ts-ignore
+        // eslint-disable-next-line react/no-this-in-sfc
+        return `${categoryDescription} <br/> ${this.y}`;
```

Comment:
there is actually a "this" when it gets called but eslint needs to be disabled here